staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/generated.proto

```diff
@@ -595,6 +599,24 @@ message ServiceReference {
   optional int32 port = 4;
 }
 
+// ValidationRule is used for expression validation.
+message ValidationRule {
+  // Rule represents the validation rule which will be evaluated by CEL.
+  // ref: https://github.com/google/cel-spec
+  // The validator will be scoped to the location of the x-kubernetes-validations extension in the schema
+  // and will use `self` to represent the scoped field name.
+  // If an object property collides with a [RESERVED keyword](https://github.com/google/cel-spec/blob/v0.6.0/doc/langdef.md#syntax), it will be escaped by prepending a _ prefix.
+  // To prevent this from causing a subsequent collision, all properties with a _ prefix will always be prefixed by __ (generally, N+1 the existing number of _s).
+  // You should escape the property by prepending a _ prefix. e.g. rule against field "_pre": {"rule": "__pre > 0"}
+  // We will exclude self and [reserved identifiers](https://github.com/google/cel-spec/blob/v0.6.0/doc/langdef.md#values) from the root bound variables without escaping.
```

Comment:
is the different link from line 608 expected?

Reply:
It is expected. line 608 is for RESERVED symbols defined in the CEL lexer.  No identifiers are allowed to collide with these symbol and we will escape with prefixed `_`.  Line 611 is for identifiers that are part of the CEL language. We do not escape with prefix `_` . Those identifiers are not allowed to be declared as root variables in cel and can be access by `self.<identifier>`.